farmfs/fs.py

```diff
@@ -43,7 +42,9 @@ def __init__(self, path, frame=None):
       assert frame is None
       self._path = path._path
     else:
-      assert isinstance(path, basestring)
+      if isinstance(path, rawtype):
+        path = raw2str(path)
+      assert isinstance(path, str), "Paths cannot be constructed from %s: %s" % (type(path), path)
       if frame is None:
```

Comment:
I think we don't need to check in path construction, other than asserts. The sources of the wrong type should be limited to 

- [ ] reading snapshots
- [ ] listdir reading the tree
- [ ] getting cwd
- [ ] command line params
- [ ] checksumming